agent/consul/leader_connect_ca.go

```diff
@@ -154,16 +154,24 @@ func (c *CAManager) setState(newState caState, validateState bool) (caState, err
 	state := c.state
 
 	if !validateState ||
-		state == caStateInitialized ||
+		(state == caStateInitialized && newState != caStateInitializing) ||
```

Comment:
nit: we should probably update the comment above that lists the allowed state transitions